sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeFormatter.scala

```diff
@@ -95,9 +95,128 @@ object CodeFormatter {
     new CodeAndComment(code.result().trim(), map)
   }
 
-  def stripExtraNewLinesAndComments(input: String): String = {
+  def stripExtraNewLinesAndCommentsUsingRegexp(input: String): String = {
     extraNewLinesRegexp.replaceAllIn(commentRegexp.replaceAllIn(input, ""), "\n")
```

Comment:
can we use a single regex to do it in one pass?

Reply:
`commentRegexp` is the source of slowness not the 2 pass.

Reply:
How about calling `commentRegexp.replaceAllIn` only when we find `/` in `input`?